utils.js

```diff
@@ -102,11 +132,9 @@ let utils = {
     wrapRequest: function wrapRequest(chakramMethod, args, options) {
         let retryVersion = options.retryVersion;
         return chakramMethod.apply(chakram, args).then((response) => {
-            if (!response || !response.response) {
-                throw new Error('Invalid response received!');
-            }
+            utils.assertSaneResponse(response);
 
-            if (response.response.statusCode === 200) {
+            if (utils.isSuccessfulResponse(response.response.statusCode)) {
```

Comment:
This is the breaking change I mentioned. I thought it more appropriate to accept any successful response.

I think we can reasonably split this out into a separate PR if desired; I don't believe anything I've implemented depends on it.